web/src/components/header/RightNav.tsx

```diff
@@ -10,14 +10,39 @@ import { Menu, Transition } from "@headlessui/react";
 import { classNames } from "@utils";
 
 import { RightNavProps } from "./_shared";
-import { Cog6ToothIcon, ArrowLeftOnRectangleIcon } from "@heroicons/react/24/outline";
-import {Link} from "@tanstack/react-router";
+
+import { Cog6ToothIcon, ArrowLeftOnRectangleIcon, MoonIcon, SunIcon } from "@heroicons/react/24/outline";
+import { Link } from "@tanstack/react-router";
+import { SettingsContext } from "@utils/Context";
 
 export const RightNav = (props: RightNavProps) => {
+  const [settings, setSettings] = SettingsContext.use();
+
+  const toggleTheme = () => {
+    setSettings(prevState => ({
+      ...prevState,
+      darkTheme: !prevState.darkTheme
+    }));
+  };
+
+  const themeIcon = settings.darkTheme;
+
+
   return (
     <div className="hidden sm:block">
       <div className="ml-4 flex items-center sm:ml-6">
-        <Menu as="div" className="ml-3 relative">
+        <div className="mt-1 items-center">
+          <button
+            onClick={toggleTheme}
+            className="p-1 rounded-full focus:outline-none focus:none transition duration-100 ease-out transform hover:bg-gray-200 dark:hover:bg-gray-800 hover:scale-100">
+            {themeIcon ? (
+              <MoonIcon className="h-4 w-4 text-gray-500 transition duration-100 ease-out transform" aria-hidden="true" />
+            ) : (
+              <SunIcon className="h-4 w-4 text-gray-600" aria-hidden="true" />
+            )}
```

Comment:
Don't really see a need here for the new variable `themeIcon`. It's a bit clearer to read `settings.darkTheme` in the button conditional for the icon.